### PR TITLE
HIP 136: Eliminate IOT Rewards for Redundant Coverage

### PR DESCRIPTION
**Pull request by No1-at-all**
When hotspots are less than 8 res 12 hexes apart only the first to witness shall be rewarded

### CODE REVIEW

**Comment by No1-at-all** [comment]
First pull

**Comment by No1-at-all** [comment]
Added explanations

**Comment by abhay** [comment]
I gave @waveform06 some recommendations for edits / simplification. Overall I think this is a great direction for IOT PoC.

**Comment by abhay** [comment]
Made some formatting changes and addressed some typos. I think the language can still be worked on a bit but no reason to hold up merging. Recommending landing this @waveform06

**Comment by waveform06** [comment]
Yes,  agreed with @No1-at-all that the language could be improved in places but I think we should get the discussion of how it affects rewards going first

File: 0136-eliminate-iot-rewards-for-redundant-coverage.md [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+3. For each witnessing Hotspot:
+   1. Iterate over the later Hotspots to determine whether they are within the *Minimum Distance* from the selected Hotspot.
+   2. If they are, mark the later Hotspots as ineligible for reward due to *Redundant Coverage*
+4. Once *Redundant Coverage* has been evaluated, also mark any Witnesses in excess of *Maximum Witnesses* as ineligible for reward.
+
+Developers are free to implement this logic in a way that optimizes for data processing as the above description implies an `O(n²)` implementation which can be inefficient. The intention of this example is to also confirm that the number of invalidated witnesses do not count towards the total number of witnesses in a PoC event.
+
+### Example map - 3 Hotspots in a resolution 11 hex with 7 rings
+
+![3 Hotspots res 11 ring 7.png](/files/0000/3-Hotspots-res-11-ring-7.png)
```
Comment (hiptron) [inline 0136-eliminate-iot-rewards-for-redundant-coverage.md:59]:
```suggestion
![3 Hotspots res 11 ring 7.png](/files/0136/3-Hotspots-res-11-ring-7.png)
```

File: 0136-eliminate-iot-rewards-for-redundant-coverage.md [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+3. For each witnessing Hotspot:
+   1. Iterate over the later Hotspots to determine whether they are within the *Minimum Distance* from the selected Hotspot.
+   2. If they are, mark the later Hotspots as ineligible for reward due to *Redundant Coverage*
+4. Once *Redundant Coverage* has been evaluated, also mark any Witnesses in excess of *Maximum Witnesses* as ineligible for reward.
+
+Developers are free to implement this logic in a way that optimizes for data processing as the above description implies an `O(n²)` implementation which can be inefficient. The intention of this example is to also confirm that the number of invalidated witnesses do not count towards the total number of witnesses in a PoC event.
+
+### Example map - 3 Hotspots in a resolution 11 hex with 7 rings
+
+![3 Hotspots res 11 ring 7.png](/files/0136/3-Hotspots-res-11-ring-7.png)
```
Comment (hiptron) [inline 0136-eliminate-iot-rewards-for-redundant-coverage.md:59]:
```suggestion
![3 Hotspots res 11 ring 7.png](/files/0136/3-hotspots-res-11-ring-7.png)
```

**Pull request merged**